### PR TITLE
WiP - WebRTCToolbox

### PR DESCRIPTION
**Pull request by kanawish**
## Description

This is our starter commit for this branch.

## Pre-merge Checklist

- [x] This code compiles/passes CI.
- [x] I've tested this fix/feature.
- [ ] I have one or more reviewer approval.

## Extra info

This is the 'baseline' of the WebRTCToolbox. The UX allows for two default 'rooms' to be accessed by 2 peers.

Here are the current Web and Mobile clients side by side.
<div style="display: flex;align-items: flex-start;">
image
image
image
</div>

Work is planned to simplify match-making between peers. See issue #2

### CODE REVIEW

File: core/kmpWebRTCToolbox/androidApp/src/main/java/com/kanastruk/webrtc/sample/ui/GamepadView.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    Column {
+        Row {
+            Spacer(Modifier.weight(1f))
+            val leftStickValues = bytes.let { gp ->
+                "L(${gp?.getOrElse(0) { 0 }},${gp?.getOrElse(1) { 0 }})"
+            } ?: "null"
+            Text(color = Color.White, text = leftStickValues)
+            Spacer(Modifier.weight(1f))
+            val rightStickValues = bytes.let { gp ->
+                "R(${gp?.getOrElse(2) { 0 }},${gp?.getOrElse(3) { 0 }})"
```
Comment (kanawish) [inline core/kmpWebRTCToolbox/androidApp/src/main/java/com/kanastruk/webrtc/sample/ui/GamepadView.kt:30]: This is a bit funky but a nice way to generate a String.

File: core/kmpWebRTCToolbox/androidApp/google-services.json.placeholder [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "TODO": "PROVISION GOOGLE SERVICE SETTINGS"
```
Comment (kanawish) [inline core/kmpWebRTCToolbox/androidApp/google-services.json.placeholder:2]: Could you pick better placeholders?

Comment (kanawish) [reply core/kmpWebRTCToolbox/androidApp/google-services.json.placeholder:2]: Oh yes! Those are terrible, let me fix that..

**Comment by smasse** [comment]
I will review tomorrow Feb 20

File: core/kmpWebRTCToolbox/androidApp/src/main/java/com/kanastruk/webrtc/sample/App.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * https://medium.com/androiddevelopers/coroutines-patterns-for-work-that-shouldnt-be-cancelled-e26c40f142ad
+ */
+class App : Application() {
+    // TODO: Check up on best practices for Android coroutine scope usage.
+    // No need to cancel this scope (for now) as it'll be torn down with the process
+    private val applicationScope = CoroutineScope(SupervisorJob())// + otherConfig)
+
+    // TODO: Injection & scope handling
+    val firebaseModel by lazy {
```
Comment (kanawish) [inline core/kmpWebRTCToolbox/androidApp/src/main/java/com/kanastruk/webrtc/sample/App.kt:20]: Oh did you consider using dependency injection?

File: core/kmpWebRTCToolbox/androidApp/src/main/java/com/kanastruk/webrtc/sample/App.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    val roomModel by lazy {
+        RoomModel(RoomFirestoreAndroid(), applicationScope)
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        if (BuildConfig.DEBUG) {
+            Timber.plant(Timber.DebugTree())
```
Comment (kanawish) [inline core/kmpWebRTCToolbox/androidApp/src/main/java/com/kanastruk/webrtc/sample/App.kt:32]:
```suggestion
            Timber.plant( Timber.DebugTree() )
```

**Review by smasse — APPROVED** [review]
saw two resolved issues, nothing major, maybe just training examples for me.

**Comment by smasse** [comment]
I assume that you will do the merge yourself, Étienne.

**Comment by smasse** [comment]
Newbie question: "...Web and Mobile clients..." are these two clients both "callers" (in the contextual terminology of "caller" and "callee", where the caller is a remote peer to the robot car which is the callee) or is one of the "clients" a caller and the other a callee?

**Pull request merged**